muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/credentials/body.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import {connect, Dispatch} from 'react-redux';
+import {i18nType} from '~/reducers/base/i18n';
+import $ from '~/lib/jquery';
+import ReturnCredentials from './body/return-credentials';
+import CredentialsContainer from './body/credentials-container';
+import CredentialsHero from './body/credentials-hero';
+import {StateType} from '~/reducers';
+
+interface CredentialsBodyProps {
+  i18n: i18nType
+}
+
+interface CredentialsBodyState {
+  
+}
+
+class CredentialsBody extends React.Component<CredentialsBodyProps, CredentialsBodyState> {
+  private param = new URLSearchParams(location.search);
```

Comment:
Pass this parameter via the props, it's nice and all as it is, but if we ever implement a router later this would fail to function; the parameter should pass as a prop, this.props.secret to this component